syntax = "proto3";
package auth;

service AuthService {
  rpc ValidateToken (ValidateRequest) returns (ValidateReply);
  rpc Login (LoginRequest) returns (AuthReply);
  rpc Register (RegisterRequest) returns (AuthReply);
}

message ValidateRequest {
  string token = 1;
}

message ValidateReply {
  bool valid = 1;
  string user_id = 2;
  string error = 3;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message RegisterRequest {
  string username = 1;
  string email = 2;
  string password = 3;
}

message AuthReply {
  string access_token = 1;
  string refresh_token = 2;
  int64 expires_at = 3;
}